src/org/omegat/gui/main/MainWindow.java

```diff
@@ -124,28 +127,32 @@ public class MainWindow extends JFrame implements IMainWindow {
     protected JLabel statusLabel;
     protected JLabel lockInsertLabel;
 
-    MainWindowAccessTools mainWindowAccessTools;
-
     protected DockingDesktop desktop;
 
     /** Creates new form MainWindow */
+    @SuppressWarnings("unchecked")
     public MainWindow() throws IOException {
         MainWindowMenuHandler mainWindowMenuHandler = new MainWindowMenuHandler(this);
-        if (Preferences.isPreference(Preferences.APPLY_BURGER_SELECTOR_UI) && !Platform.isMacOSX()) {
-            menu = new MainWindowBurgerMenu(this, mainWindowMenuHandler);
-            menu.initComponents();
-            mainWindowAccessTools = MainWindowAccessTools.of(menu.mainMenu, mainWindowMenuHandler);
+
+        // Load Menu extension
+        Class<? extends BaseMainWindowMenu> menuClass;
+        menuClass = (Class<? extends BaseMainWindowMenu>) UIManager.get(menuClassID);
+        if (menuClass != null) {
+            try {
+                menu = menuClass.getDeclaredConstructor(MainWindow.class, MainWindowMenuHandler.class)
+                                .newInstance(this, mainWindowMenuHandler);
+            } catch (InstantiationException | IllegalAccessException | InvocationTargetException |
+                     NoSuchMethodException ignored) {
+                // fall back to default when loading failed.
+                menu = new MainWindowMenu(this, mainWindowMenuHandler);
+            }
```

Comment:
To enable the final variable I propose this:
```suggestion
        if (menuClass != null) {
            BaseMainWindowMenu menu1;
            try {
                menu1 = menuClass.getDeclaredConstructor(MainWindow.class, MainWindowMenuHandler.class)
                                .newInstance(this, mainWindowMenuHandler);
            } catch (InstantiationException | IllegalAccessException | InvocationTargetException |
                     NoSuchMethodException ignored) {
                // fall back to default when loading failed.
                menu1 = new MainWindowMenu(this, mainWindowMenuHandler);
            }
            menu = menu1;
        } else {
```